fuzzing/replay/replay_main.cc

```diff
@@ -22,7 +22,7 @@
 
 extern "C" {
 int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size);
-int LLVMFuzzerInitialize(int* argc, char*** argv) __attribute__((weak_import));
+int LLVMFuzzerInitialize(int* argc, char*** argv) __attribute__((weak));
```

Comment:
It appears that here the code is reverting to the state before #139 - which was problematic because it was breaking MacOS builds. Can you confirm there are no regressions on this front?

Reply:
I think that the relevant change for macOS compatibility was the explicit undefined declaration for `LLVMFuzzerInitialize`, not the change from `weak` to `weak_import` (see https://stackoverflow.com/a/34983229/297261). I don't have access to a Mac at the moment to verify this myself, but added another commit that tests whether the replay engine works on macOS.

Reply:
https://github.com/bazelbuild/rules_fuzzing/pull/177/checks?check_run_id=3354656650 shows that the replay launcher compiles on macOS with the changes from this PR. The regression tests still fail since the launcher scripts rely on `readlink -f`, which is unavailable on macOS, but that is an unrelated issue.

Reply:
Ugh, and I couldn't find an easy replacement unfortunately.

The best I could come with is replacing `readlink -f "<path>"` with `python -c 'import os, sys; print(os.path.realpath(sys.argv[1]))' "<path>"`. If you get the chance, can you test out this change to see if it makes a difference?

Reply:
That will make the script less compatible with various Linux distributions that by default offer `python3` but not `python`. An alternative could be to ask users to install [`coreutils`](https://formulae.brew.sh/formula/coreutils) and fall back to `greadlink -f` if `readlink -f` fails.

Reply:
> That will make the script less compatible with various Linux distributions that by default offer `python3` but not `python`.

Should we use `python3` then, instead? I believe the fuzzing rules anyway have a hard dependency on it through the fuzzer launcher.

Reply:
That's a good point, made the change and it seems to work.